lib/core/macros.nim

```diff
@@ -1697,3 +1697,26 @@ proc extractDocCommentsAndRunnables*(n: NimNode): NimNode =
         result.add ni
       else: break
     else: break
+
+proc freshIdentNodes*(n: NimNode): NimNode =
+  ## Replaces every `nnkSym` node in `n` by a fresh identifier node.
+  ## This forces the compiler to perform a new lookup pass.
+  runnableExamples:
+    macro lc(init: untyped): untyped =
+      let x = freshIdentNodes(init)
```

Comment:
```nim
      expectKind(init, nnkSym) or doAssert ...
```

Reply:
I can do that but init can be in {nnkIdent, nnkSym} or anything else really. Should I try to test the compiler?

Reply:
I think the example should show that the NimNode is not `nnkIdent` and after `freshIdentNodes`, it becomes `nnkIdent`.

Reply:
done.